monkey/tests/unit_tests/infection_monkey/island_api_client/test_http_island_api_client.py

```diff
@@ -441,23 +444,34 @@ def test_island_api_client__get_credentials_for_propagation(
         (501, IslandAPIRequestFailedError),
     ],
 )
-def test_island_api_client_get_credentials_for_propagation__status_code(
+def test_island_api_client_get_agent_signals__status_code(
     island_api_client, status_code, expected_error
 ):
     with requests_mock.Mocker() as m:
         m.get(ISLAND_URI)
         island_api_client.connect(SERVER)
 
         with pytest.raises(expected_error):
-            m.get(ISLAND_GET_PROPAGATION_CREDENTIALS_URI, status_code=status_code)
-            island_api_client.get_credentials_for_propagation()
+            m.get(ISLAND_GET_AGENT_SIGNALS, status_code=status_code)
+            island_api_client.get_agent_signals(agent_id=AGENT_ID)
 
 
-def test_island_api_client_get_credentials_for_propagation__bad_json(island_api_client):
+def test_island_api_client_get_agent_signals(island_api_client):
     with requests_mock.Mocker() as m:
         m.get(ISLAND_URI)
         island_api_client.connect(SERVER)
 
-        with pytest.raises(IslandAPIRequestFailedError):
-            m.get(ISLAND_GET_PROPAGATION_CREDENTIALS_URI, content=b"bad")
-            island_api_client.get_credentials_for_propagation()
+        m.get(ISLAND_GET_AGENT_SIGNALS, json={"terminate": TIMESTAMP})
+        actual_terminate_timestamp = island_api_client.get_agent_signals(agent_id=AGENT_ID)
+
+        assert actual_terminate_timestamp == TIMESTAMP
```

Comment:
Should we have a test for the `None` return as well (since the timestamp is optional)?

Reply:
I will parametrize. Thanks!